[Demo] Add waterfall chart to visual-vocabulary (#786)
mckinsey/vizro: vizro-core/examples/visual-vocabulary/README.md

| Treemap               | ✅     | Part-to-whole            | [Treemap with px](https://plotly.com/python/treemaps/)                                     | [px.treemap](https://plotly.com/python-api-reference/generated/plotly.express.treemap.html)                                                                                                     |
| Venn                  | ❌     | Part-to-whole            |                                                                                            |                                                                                                                                                                                                 |
| Violin                | ✅     | Distribution             | [Violin plot with px](https://plotly.com/python/violin/)                                   | [px.violin](https://plotly.com/python-api-reference/generated/plotly.express.violin.html)                                                                                                       |

> | Waterfall             | ✅     | Part-to-whole, Flow      | [Waterfall with px](https://plotly.com/python/waterfall-charts/)                           |                                                                                                                                                                                                 | [go.Waterfall](https://plotly.github.io/plotly.py-docs/generated/plotly.graph_objects.Waterfall.html) |

Reviewer: ```suggestion
| Waterfall             | ✅     | Part-to-whole, Flow      | [Waterfall charts in Python](https://plotly.com/python/waterfall-charts/)                           |                                                                                                                                                                                                 | [go.Waterfall](https://plotly.github.io/plotly.py-docs/generated/plotly.graph_objects.Waterfall.html) |
```

Note it's not available in plotly.express (`px`), only in `go`.

> | Waterfall             | ✅     | Part-to-whole, Flow      | [Waterfall charts in Python](https://plotly.com/python/waterfall-charts/)                  |                                                                                                                                                                                                 | [go.Waterfall](https://plotly.github.io/plotly.py-docs/generated/plotly.graph_objects.Waterfall.html) |

Reviewer 2: To stay consistent with the phrasing of charts that use `go` charts and to give credits to Plotly :)

```suggestion
| Waterfall             | ✅     | Part-to-whole, Flow      | [Waterfall charts in Plotly](https://plotly.com/python/waterfall-charts/)                  |                                                                                                                                                                                                 | [go.Waterfall](https://plotly.github.io/plotly.py-docs/generated/plotly.graph_objects.Waterfall.html) |
```
